demo/src/test/java/com/mongodb/jdbc/utils/TestUtils.java

```diff
@@ -0,0 +1,439 @@
+
+        class QueryExecutor implements Runnable
+        {
+            private boolean m_isExecuting;
+            private final String m_query;
+            private ResultSet m_result;
+            private final Statement m_statement;
+
+            public QueryExecutor(Statement statement, String query)
+            {
+                m_statement = statement;
+                m_query = query;
+                m_isExecuting = false;
+            }
+
+            public boolean getIsExecuting()
+            {
+                return m_isExecuting;
+            }
+
+            public ResultSet getResultSet()
+            {
+                return m_result;
+            }
+
+            @Override
+            public void run()
+            {
+                try
+                {
+                    System.out.println("Executing '" + m_query + "'");
+                    m_isExecuting = true;
+                    m_result = m_statement.executeQuery(m_query);
+                    m_isExecuting = false;
+                }
+                catch (SQLException e)
+                {
+                    e.printStackTrace();
+                }
+                finally {
+                    try {
+                        m_result.close();
+                    } catch (SQLException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+
+        Statement statement = m_conn.createStatement();
+        ResultSet result = null;
+
+        QueryExecutor queryExecutor = new QueryExecutor(statement, query);
+        Thread execThread = new Thread(queryExecutor);
+        execThread.start();
+
+        while (!queryExecutor.getIsExecuting())
+        {
+            ;
+        }
+        if (execThread.isAlive())
+        {
+            try
+            {
+                // Still need to wait because the m_isExecuting flag is set  before statement.executeQuery is called.
+                // Need to wait until an executor object is created.
+                Thread.sleep(2000);
+            }
+            catch (InterruptedException e)
+            {
+                e.printStackTrace();
+            }
+            System.out.println("Calling cancel");
+            statement.cancel();
+            System.out.println("The query cancelled successfully.");
+
+            // Try to use the statement again
+            queryExecutor = new QueryExecutor(statement, query);
+            execThread = new Thread(queryExecutor);
+            execThread.start();
+        }
+        else
+        {
+            System.out.println("The execution finished before cancel");
+            queryExecutor.getResultSet().close();
+        }
+
+        System.out.println("-----------------------------------------------------------------------");
+    }
+
+    private void runExecuteQuery(String query, boolean printRs) throws Exception
```

Comment:
```suggestion
    // Run executeQuery and print the result set metadata. Optionally print the result set data by arguing true as the second parameter.
    private void runExecuteQuery(String query, boolean printRs) throws Exception
```
A briefly comment explaining what this does would be helpful. Especially if we follow-up on Oliver's idea!